app/controllers/albums_controller.rb

```
@@ -0,0 +1,62 @@
+class AlbumsController < ApplicationController
+
+  before_action :get_album, only: [:show, :edit, :update, :upvote, :destroy]
+
+  def get_album
+    @album = Album.find(params[:id])
+  end
+
+  # Show album(s)
+  def index
+    @albums = Album.order(:name)
+  end
+
+  def show
+  end
+
+  # Add an album
+  def new
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.new(album_params)
+    if @album.save
+      redirect_to album_path(@album.id)
+    else
+      render :new
+    end
+  end
+
+  # Edit an album
+  def edit
+  end
+
+  def update
+    @album.update(album_params)
+
+    redirect_to album_path
+  end
+
+  # Upvote an album
+  def upvote
+    @album.vote += 1
+    @album.save
+
+    redirect_to :back rescue redirect_to album_path
```

Avoid doing blind rescues when possible. Catching the specific error you're expecting to be thrown is often best practice, as catching all errors can hide other errors that are happening.